bzt/modules/apiritif/generator.py

```diff
@@ -630,12 +635,43 @@ def _gen_action(self, action_config):
             action_elements.extend(self._gen_alert(param))
         elif atype == 'if':
             action_elements.append(self._gen_condition_mngr(param, action_config))
+        elif atype == 'loop':
+            action_elements.append(self._gen_loop_mngr(action_config))
 
         if not action_elements and not self.ignore_unknown_actions:
             raise TaurusInternalException("Could not build code for action: %s" % action_config)
 
         return [ast.Expr(element) for element in action_elements]
 
+    def _gen_loop_mngr(self, action_config):
+        if 'start' not in action_config or 'end' not in action_config or 'do' not in action_config:
+            raise TaurusConfigError("Loop must contain start, end and do")
+        elements = []
+        range_elts = []
+        start = action_config['start']
+        end = action_config['end']
+        step = action_config['step'] or 1
```

Comment:
please make some refactoring there according to our routine. I prepared [refactoring of conditions](https://github.com/Blazemeter/taurus/pull/1272) as example. It should look like
```
exc = TaurusConfigError("Loop must contain start, end and do")
start = action_config.get("start", exc)
end = ...
do = ...
step = action_config.get("step", 1)
```

Reply:
Agreed, this is definitely looking much better.